src/routes/safe/components/Transactions/TxsTable/ExpandedTx/OwnersColumn/index.jsx

```diff
@@ -7,86 +7,118 @@ import Block from '~/components/layout/Block'
 import Col from '~/components/layout/Col'
 import Img from '~/components/layout/Img'
 import { type Owner } from '~/routes/safe/store/models/owner'
-import { type Transaction } from '~/routes/safe/store/models/transaction'
+import { makeTransaction, type Transaction } from '~/routes/safe/store/models/transaction'
 import { TX_TYPE_CONFIRMATION } from '~/logic/safe/transactions/send'
-import { EMPTY_DATA } from '~/logic/wallets/ethTransactions'
 import OwnersList from './OwnersList'
 import ButtonRow from './ButtonRow'
 import CheckLargeFilledGreenIcon from './assets/check-large-filled-green.svg'
 import ConfirmLargeGreenIcon from './assets/confirm-large-green.svg'
+import CheckLargeFilledRedIcon from './assets/check-large-filled-red.svg'
+import ConfirmLargeRedIcon from './assets/confirm-large-red.svg'
 import ConfirmLargeGreyIcon from './assets/confirm-large-grey.svg'
 import { styles } from './style'
 import Paragraph from '~/components/layout/Paragraph/index'
 
 type Props = {
   tx: Transaction,
+  cancelTx: Transaction,
   owners: List<Owner>,
   classes: Object,
   granted: boolean,
   threshold: number,
   userAddress: string,
   thresholdReached: boolean,
-  safeAddress: string,
+  cancelThresholdReached: boolean,
   canExecute: boolean,
   onTxConfirm: Function,
+  onCancelTxConfirm: Function,
   onTxCancel: Function,
-  onTxExecute: Function
+  onTxExecute: Function,
+  onCancelTxExecute: Function,
+  canExecuteCancel: boolean,
 }
 
-const isCancellationTransaction = (tx: Transaction, safeAddress: string) => !tx.value && tx.data === EMPTY_DATA && tx.recipient === safeAddress
-
-const OwnersColumn = ({
-  tx,
-  owners,
-  classes,
-  granted,
-  threshold,
-  userAddress,
-  thresholdReached,
-  safeAddress,
-  onTxConfirm,
-  onTxCancel,
-  onTxExecute,
-  canExecute,
-}: Props) => {
-  const cancellationTx = isCancellationTransaction(tx, safeAddress)
-  const showOlderTxAnnotation = thresholdReached && !canExecute && !tx.isExecuted
-
+function ownersConfirmations(tx = makeTransaction(), userAddress) {
```

Comment:
It's a good practice to start function name with a verb, so let's rename it to `getOwnersConfirmations`